spec/linter-registry-spec.coffee

```diff
@@ -0,0 +1,169 @@
+describe 'linter-registry', ->
+  LinterRegistry = require('../lib/linter-registry')
+  EditorLinter = require('../lib/editor-linter')
+  beforeEach ->
+    waitsForPromise ->
+      atom.workspace.destroyActivePaneItem()
+      atom.workspace.open('test.txt')
+  describe '::construct', ->
+    it 'expects no arguments', ->
+      linterRegistry = new LinterRegistry
+      expect(true).toBe(true)
+      linterRegistry.deactivate()
+  describe '::addLinter', ->
+    it 'adds error notification if linter is invalid', ->
+      linterRegistry = new LinterRegistry
+      linterRegistry.addLinter({})
+      linterRegistry.deactivate()
+      expect(atom.notifications.getNotifications().length).toBe(1)
+    it 'pushes linter into registry when valid', ->
+      linter = {grammarScopes: ['*'], lintOnFly: false, modifiesBuffer: false, scope: 'file', lint: -> }
+      linterRegistry = new LinterRegistry
+      linterRegistry.addLinter(linter)
+      expect(linterRegistry.linters.length).toBe(1)
+      linterRegistry.deactivate()
+  describe '::hasLinter', ->
+    it 'returns true if present', ->
+      linter = {grammarScopes: ['*'], lintOnFly: false, modifiesBuffer: false, scope: 'file', lint: -> }
+      linterRegistry = new LinterRegistry
+      linterRegistry.addLinter(linter)
+      expect(linterRegistry.hasLinter(linter)).toBe(true)
+      linterRegistry.deactivate()
+    it 'returns false if not', ->
+      linter = {grammarScopes: ['*'], lintOnFly: false, modifiesBuffer: false, scope: 'file', lint: -> }
+      linterRegistry = new LinterRegistry
+      expect(linterRegistry.hasLinter(linter)).toBe(false)
+      linterRegistry.deactivate()
+  describe '::deleteLinter', ->
+    it 'deletes the linter from registry', ->
+      linter = {grammarScopes: ['*'], lintOnFly: false, modifiesBuffer: false, scope: 'file', lint: -> }
```

Comment:
this generic linter is used multiple times, you should create a named function to build it so that you don't repeat code and it's specific definition is better locked for people reading the tests.